app/src/org/commcare/heartbeat/HeartbeatWorker.kt

```diff
@@ -0,0 +1,30 @@
+package org.commcare.heartbeat
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.util.LogTypes
+import org.javarosa.core.services.Logger
+
+/**
+ * @author $|-|!˅@M
+ */
+class HeartbeatWorker(context: Context, workerParams: WorkerParameters):
+        Worker(context, workerParams) {
+    private val requester = CommCareApplication.instance().heartbeatRequester
+    override fun doWork(): Result {
+        try {
+            if (!CommCareApplication.instance().session.heartbeatSucceededForSession()) {
```

Comment:
we should also have a `session.isActive` check here to make sure we don't fire a heartbeat after the session has expired.